sql/analyzer/declare.go

```diff
@@ -134,11 +337,65 @@ func resolveDeclarationsInner(ctx *sql.Context, a *Analyzer, node sql.Node, scop
 			}
 			newChild = plan.NewSignal(condition.SqlStateValue, c.Signal.Info)
 			same = false
+		case *plan.Open:
+			cursor := scope.GetCursor(c.Name)
+			if cursor == nil {
+				return nil, transform.SameTree, sql.ErrCursorNotFound.New(c.Name)
+			}
+			newChild = c.WithId(cursor.id)
+			same = false
+		case *plan.Close:
+			cursor := scope.GetCursor(c.Name)
+			if cursor == nil {
+				return nil, transform.SameTree, sql.ErrCursorNotFound.New(c.Name)
+			}
+			newChild = c.WithId(cursor.id)
+			same = false
+		case *plan.Fetch:
+			cursor := scope.GetCursor(c.Name)
+			if cursor == nil {
+				return nil, transform.SameTree, sql.ErrCursorNotFound.New(c.Name)
+			}
+			newChild, _, err = resolveProcedureVariables(ctx, scope, c.WithId(cursor.id))
+			if err != nil {
+				return nil, transform.SameTree, err
+			}
+			same = false
+		case *plan.InsertInto:
+			var newSource sql.Node
+			newSource, same, err = resolveProcedureVariables(ctx, scope, c.Source)
+			if err != nil {
+				return nil, transform.SameTree, err
+			}
+			if same == transform.NewTree {
+				newChild = c.WithSource(newSource)
+			}
+		case *plan.Union:
+			// todo(max): IndexedJoins might be missed here
```

Comment:
?

Reply:
If this is a real issue it should have a skipped test

Reply:
TBH not 100% sure what the comment is for, this is something that Max did. I didn't edit any of this part, I just moved it from the other file when I consolidated the logic.

Reply:
@max-hoffman Do you know if we have a test for this?

Reply:
i think i wrote this because indexJoins did not implement Expressioner to expose its join condition until a couple months ago. i did not include tests as part of rewriting all the transformation rules, so probably not.